rtrlib/transport/ssh/ssh_transport.c

```diff
@@ -290,46 +290,91 @@ RTRLIB_EXPORT int tr_ssh_init(const struct tr_ssh_config *config, struct tr_sock
 	socket->send_fp = &tr_ssh_send;
 	socket->ident_fp = &tr_ssh_ident;
 
-	socket->socket = lrtr_malloc(sizeof(struct tr_ssh_socket));
+	socket->socket = lrtr_calloc(1, sizeof(struct tr_ssh_socket));
 	struct tr_ssh_socket *ssh_socket = socket->socket;
 
 	ssh_socket->channel = NULL;
 	ssh_socket->session = NULL;
 	ssh_socket->config.host = lrtr_strdup(config->host);
+	if (!ssh_socket->config.host)
+		goto error;
 	ssh_socket->config.port = config->port;
+
 	ssh_socket->config.username = lrtr_strdup(config->username);
+	if (!ssh_socket->config.username)
+		goto error;
 
```

Comment:
move it at same level as config.host. it avoids strdupping (config->username) for notthing.

Reply:
You are saying I should check if username is set before strduping it?